functions/index.js

```diff
@@ -143,3 +143,183 @@ app.post('/', isAuthenticated ,function(req,response)
 });
 
 exports.signUp = functions.https.onRequest(app);
+
+
+
+
+
+const db = admin.database().ref();
+
+exports.addEvent = functions.https.onRequest((req,res) => {
+
+    let eventCategory = req.body.eventCategory;
+    let eventName = req.body.eventName;
+    let startTime = req.body.startTime;
+    let endTime = req.body.endTime;
+    let eventDescription = req.body.eventDescription;
+
+    db.child(`events/${eventCategory}/${eventName}`).set({
+
+        name : eventName,
+        startTime : startTime,
+        endTime : endTime
+    })
+    .then((snapshot) => {
+        console.log('done')
+    }).catch((err) => {
+        res.send(err)
+    })
+
+    db.child(`eventDescription/${eventCategory}/${eventName}`).set({
+
+        name : eventName,
+        eventDescription : eventDescription,
+        startTime : startTime,
+        endTime : endTime
+    }).then((snapshot) => {
+        return res.send(snapshot.val())
+    }).catch((err) => {
+        return res.send(err)
+    }) 
+})
+
+exports.getCategories = functions.https.onRequest((req,res) => {
+
+    return db.child('events').once('value')
+    .then((snapshot) => {
+        
+        var data = {categories : []}
+        for(var i in snapshot.val())
+        {
+            var obj = {};
+            obj.name = i;
+            data.categories.push(obj);
+        }
+        return res.send(data);
+    })
+    .catch((err) => {
+        return res.send(err);
+    })
+})
+
+exports.getEventNames = functions.https.onRequest((req,res) => {
+
+    if(req.query.category == 'all')
+    {
+        return db.child('events').once('value')
+        .then((snapshot) => {
+            
+            var data = {events : []}
+            var database=snapshot.val();
+            for(var category in database)
+            {
+                for(let event in database[category])
+                {
+                    var obj = {};
+                    obj.category = category,
+                    obj.name = database[category][event].name;
```

Comment:
Change this as previous: {
	"managerial": ["a", "b"],
        "programming" : ["a","b"]
}